frontend/src/main/web/lib/utils/GlossaryHelper.js

```diff
@@ -0,0 +1,115 @@
+import _ from 'lodash';
+import StringUtils from './StringUtils'
+
+var GlossaryHelper = {
+  /**
+   * Generate org.zanata.rest.dto.GlossaryTerm object
+   * @param data
+   */
+  generateGlossaryTermDTO: function (data) {
+    if(_.isUndefined(data)) {
+      return null;
+    }
+    var content = StringUtils.trim(data.content),
+      locale = data.locale,
+      comments = StringUtils.trim(data.comment);
+
+    if(StringUtils.isEmptyOrNull(content) &&
+        StringUtils.isEmptyOrNull(locale)) {
+      return null;
+    } else {
+      var term = {};
+      term['content'] = content;
+      term['locale'] = locale;
+      term['comments'] = comments;
+      return term;
+    }
+  },
+
+  /**
+   * Generate org.zanata.rest.dto.Glossary object
+   * @param data
+   */
+  generateGlossaryDTO: function (data) {
+    var glossary = {}, entry = {};
+    glossary['glossaryEntries'] = [];
+
+    entry['resId'] = data.resId;
+    entry['pos'] = StringUtils.trim(data.pos);
+    entry['description'] = StringUtils.trim(data.description);
+    entry['srcLang'] = data.srcTerm.locale;
+    entry['sourceReference'] = data.srcTerm.reference;
+    entry['glossaryTerms'] = [];
+
+    var srcTerm = this.generateGlossaryTermDTO(data.srcTerm);
+    if(!_.isNull(srcTerm)) {
+      entry['glossaryTerms'].push(srcTerm);
+    }
+
+    var transTerm = this.generateGlossaryTermDTO(data.transTerm);
+    if(!_.isNull(transTerm)) {
+      entry['glossaryTerms'].push(transTerm);
+    }
+
+    glossary['glossaryEntries'].push(entry);
+    return glossary;
+  },
+
+  generateTerm: function(transLocaleId){
+    return {
+      content: '',
+      locale: transLocaleId,
+      comment: '',
+      lastModifiedDate: '',
+      lastModifiedBy: ''
+    }
+  },
+
+  generateSrcTerm: function (localeId) {
+    var term = this.generateTerm(localeId);
+    term['reference'] = '';
+    return term;
+  },
+
+  getTermByLocale: function (terms, localeId) {
+    var term =  _(terms)
+      .filter(function(term) { return term.locale === localeId; })
+      .value();
```

Comment:
lodash chain syntax can be more concise when there are multiple steps, but here I think it would be simpler just to use filter directly:

```
var term = _.filter(terms, term => term.locale === localeId)
```